python/paddle/v2/framework/layer.py

```diff
@@ -0,0 +1,154 @@
+from paddle.v2.framework.graph import g_program, Variable
+import paddle.v2.framework.core as core
+import copy
+import itertools
+
+
+def unique_name(prefix):
+    uid = core.unique_integer()  # unique during whole process.
+    return "_".join([prefix, str(uid)])
+
+
+def fc_layer(input,
+             size,
+             param_attr=None,
+             bias_attr=False,
+             name=None,
+             act=None,
+             num_flatten_dims=1):
+    # get name
+    if name is None:
+        name = unique_name("fc")
+
+    # check size is a integer
+    if isinstance(size, int):
+        raise TypeError("size of fc layer must be integer")
+
+    # FC accept multiple input
+    if isinstance(input, Variable):
+        input = [input]
+
+    input_type_error = TypeError(
+        "input of fc should be a variable, or a sequence of variable")
+    if not isinstance(input, list) and not isinstance(input, tuple):
+        raise input_type_error
+    else:
+        dtype = None
+        for each_var in input:
+            if not isinstance(each_var, Variable):
+                raise input_type_error
+            if dtype is None:
+                dtype = each_var.data_type
+            elif dtype != each_var.data_type:
+                raise ValueError("input of fc layer should be same data type")
+
+    # Set default value of param_attr
+    if param_attr is None:
+        param_attr = {
+            "name": None,
+            "init_attr": {
+                "type": 'uniform_random',
+                'min': -1.0,
+                'max': 1.0
+            }
+        }
+
+    if isinstance(param_attr, dict):
+        param_attr = [param_attr]
+
+    if not isinstance(param_attr, list) and not isinstance(param_attr, tuple):
+        raise TypeError(
+            "param attribute of fc should be a dict, or a sequence of dict")
```

Comment:
if `param_attr` is a list of non-dict type, it will also pass all the above checks.

Reply:
Done.